docs/proposals/proposal-trigger-and-deploy.md

```diff
@@ -0,0 +1,247 @@
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"main", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+See [design details](#design-details) section for more information.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this proposal is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Project maintainer adds green reviews pipeline to their CI/CD
+
+Participating CNCF projects will add calling the webhook to their CI/CD pipeline.
+They can use either the curl command we provide or an alternative of their choice.
+
+The fine grained access token we provide will need to be stored as a secret in
+their CI/CD pipeline.
+
+#### Project maintainer deploys their project so it can be measured
+
+Participating CNCF projects will deploy their project using a gitops approach
+with flux. This is decribed in more detail in the design details section.
+
+#### Project maintainer triggers pipeline to measure a release
+
+Calling the webhook will trigger the pipeline. The Report stage will provide
+the results to users of the project.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate?
+Think broadly.  For example, consider how this will impact or be impacted
+by scaling to support more CNCF Projects.
+
+How will this affect the benchmark tests, CNCF Project Maintainers, pipeline maintainers, etc?
+-->
+
+Multiple deployments will produce inaccurate results as we can only measure
+a single project per node. We can set concurrency in the workflow to ensure
+only a single instance runs at a time.
+
+Deployment may fail. What alerting do we need? Do we also need to notify the
+project?
+
+Uninstall at end of pipeline fails. We can wait till all flux finalizers are
+removed. In future we could create nodes on demand and delete on completion.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include manifests or workflow examples
+(though not always required) or even code snippets. If there's any ambiguity
+about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+### Trigger
+
+CNCF projects will be given a GitHub webhook they can call to trigger the
+green reviews pipeline. The webhook will accept 3 inputs and trigger a GitHub
+Actions workflow.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
+they wish to test e.g. `modern-ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
```

Comment:
@nikimanoledaki @leonardpahlke @AntonioDiTuri @locomundo At this point the design has a dependency on the other proposals. 

As any inputs needed for the benchmark and report steps will also need to be added here. 

It would be good to define these early so we can present a complete solution to the Falco team.

Reply:
As we agreed during the meeting this will be a live document - let's edit this (if needed) after proposal 2 and 3 are finalized